### PR TITLE
DM-42762: Make ApPipe contracts more human-friendly

### PR DESCRIPTION
**Pull request by kfindeisen**
This PR fixes missing `filterDiaSrcCat` in the secondary pipelines (a pipeline-breaking bug since the default connections for `rbClassify` and `transformDiaSrcCat` assume it), and then adds a `msg` argument to each connection contract.

### CODE REVIEW

File: pipelines/_ingredients/ApPipeWithFakes.yaml [hunk 1]
```diff
@@ -106,44 +113,67 @@ subsets:
...
+    msg: "subtractImagesWithFakes.difference != detectAndMeasureWithFakes.difference"
+  - contract: subtractImagesWithFakes.connections.ConnectionsClass(config=subtractImagesWithFakes).science.name ==
+              detectAndMeasureWithFakes.connections.ConnectionsClass(config=detectAndMeasureWithFakes).science.name
+    msg: "subtractImagesWithFakes.science != detectAndMeasureWithFakes.science"
+  - contract: subtractImagesWithFakes.connections.ConnectionsClass(config=subtractImagesWithFakes).template.name ==
+              diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
+    msg: "subtractImagesWithFakes.template != diaPipe.template"
+  - contract: subtractImagesWithFakes.connections.ConnectionsClass(config=subtractImagesWithFakes).science.name ==
+              diaPipe.connections.ConnectionsClass(config=diaPipe).exposure.name
+    msg: "subtractImagesWithFakes.science != detectAndMeasureWithFakes.exposure"
```
Comment (kfindeisen) [inline pipelines/_ingredients/ApPipeWithFakes.yaml:148]:
```suggestion
    msg: "subtractImagesWithFakes.science != diaPipe.exposure"
```

File: pipelines/_ingredients/ApPipeWithFakes.yaml [hunk 2]
```diff
@@ -106,44 +113,67 @@ subsets:
...
+  - contract: detectAndMeasureWithFakes.connections.ConnectionsClass(config=detectAndMeasureWithFakes).diaSources.name ==
+              filterDiaSrcCatWithFakes.connections.ConnectionsClass(config=filterDiaSrcCatWithFakes).diaSourceCat.name
+    msg: "detectAndMeasureWithFakes.diaSource != filterDiaSrcCatWithFakes.diaSourceCat"
```
Comment (kfindeisen) [inline pipelines/_ingredients/ApPipeWithFakes.yaml:151]:
```suggestion
    msg: "detectAndMeasureWithFakes.diaSources != filterDiaSrcCatWithFakes.diaSourceCat"
```

File: pipelines/_ingredients/ApPipeCalibrateImage.yaml [hunk 3]
```diff
@@ -66,43 +71,63 @@ subsets:
...
+  - contract: filterDiaSrcCat.connections.ConnectionsClass(config=filterDiaSrcCat).filteredDiaSourceCat.name ==
+              rbClassify.connections.ConnectionsClass(config=rbClassify).diaSources.name
+    msg: "filterDiaSrcCat.filteredDiaSourceCat != rbClassify.diaSources"
+  - contract: filterDiaSrcCat.connections.ConnectionsClass(config=filterDiaSrcCat).filteredDiaSourceCat.name ==
+              transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceCat.name
+    msg: "filterDiaSrcCat.filteredDiaSourceCat != transformDiaSrcCat.diaSourceCat"
+  - contract: (not transformDiaSrcCat.doIncludeReliability) or
+              (rbClassify.connections.ConnectionsClass(config=rbClassify).classifications.name ==
+               transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).reliability.name)
+    msg: "rbClassify.classifications != transformDiaSrcCat.reliability"
```
Comment (kherner) [inline pipelines/_ingredients/ApPipeCalibrateImage.yaml:130]: Might want to change the message a bit to allow for the first case (not transformDiaSrcCat.doIncludeReliability) to be the trigger as opposed to only the second case. This comment also goes for the other instances of it.

File: pipelines/_ingredients/ApPipe.yaml [hunk 4]
```diff
@@ -72,32 +72,46 @@ subsets:
...
+  - contract: filterDiaSrcCat.connections.ConnectionsClass(config=filterDiaSrcCat).filteredDiaSourceCat.name ==
+              rbClassify.connections.ConnectionsClass(config=rbClassify).diaSources.name
+    msg: "filterDiaSrcCat.filteredDiaSourceCat != rbClassify.diaSources"
+  - contract: filterDiaSrcCat.connections.ConnectionsClass(config=filterDiaSrcCat).filteredDiaSourceCat.name ==
+              transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceCat.name
+    msg: "filterDiaSrcCat.filteredDiaSourceCat != transformDiaSrcCat.diaSourceCat"
+  - contract: (not transformDiaSrcCat.doIncludeReliability) or
+              (rbClassify.connections.ConnectionsClass(config=rbClassify).classifications.name ==
+               transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).reliability.name)
+    msg: "rbClassify.classifications != transformDiaSrcCat.reliability"
```
Comment (kherner) [inline pipelines/_ingredients/ApPipe.yaml:114]: See the other comment about changing the message.

File: pipelines/_ingredients/ApPipeWithFakes.yaml [hunk 5]
```diff
@@ -106,44 +113,67 @@ subsets:
...
+  - contract: filterDiaSrcCatWithFakes.connections.ConnectionsClass(config=filterDiaSrcCatWithFakes).filteredDiaSourceCat.name ==
+              rbClassifyWithFakes.connections.ConnectionsClass(config=rbClassifyWithFakes).diaSources.name
+    msg: "filterDiaSrcCatWithFakes.filteredDiaSourceCat != rbClassifyWithFakes.diaSources"
+  - contract: filterDiaSrcCatWithFakes.connections.ConnectionsClass(config=filterDiaSrcCatWithFakes).filteredDiaSourceCat.name ==
+              transformDiaSrcCatWithFakes.connections.ConnectionsClass(config=transformDiaSrcCatWithFakes).diaSourceCat.name
+    msg: "filterDiaSrcCatWithFakes.filteredDiaSourceCat != transformDiaSrcCatWithFakes.diaSourceCat"
+  - contract: (not transformDiaSrcCatWithFakes.doIncludeReliability) or
+              (rbClassifyWithFakes.connections.ConnectionsClass(config=rbClassifyWithFakes).classifications.name ==
+               transformDiaSrcCatWithFakes.connections.ConnectionsClass(config=transformDiaSrcCatWithFakes).reliability.name)
+    msg: "rbClassifyWithFakes.classifications != transformDiaSrcCatWithFakes.reliability"
```
Comment (kherner) [inline pipelines/_ingredients/ApPipeWithFakes.yaml:173]: See the other comment about changing the message.

File: pipelines/_ingredients/ApPipeCalibrateImage.yaml (hunk 3 above)
Comment (kfindeisen) [reply pipelines/_ingredients/ApPipeCalibrateImage.yaml:130]: But if `doIncludeReliability` isn't set, then `transformDiaSrcCat.reliability` doesn't even exist (and wouldn't be used if it did), so the contract is trivially satisfied.

**Review by kherner — APPROVED** [review]
Only comments were on the message for the contracts that had (not transformDiaSrcCatWithFakes.doIncludeReliability) in them.

**Pull request merged**